gui/integrity-dialog.cpp

```diff
@@ -146,6 +146,35 @@ IntegrityDialog::IntegrityDialog(Common::String endpoint, Common::String domain)
 IntegrityDialog::~IntegrityDialog() {
 }
 
+
+bool IntegrityDialog::progressUpdate(int bytesProcessed) {
+	if (g_checksum_state->dialog->_close)
+		return false;
+	
+	g_checksum_state->calculatedSize += bytesProcessed;
+
+	if (g_system->getMillis() > g_checksum_state->lastUpdate + 500) {
+		g_checksum_state->lastUpdate = g_system->getMillis(); 
+		g_checksum_state->dialog->sendCommand(kDownloadProgressCmd, 0);
+	}
+
+	Common::Event event;
+	if (g_system->getEventManager()->pollEvent(event)) {
+		if (g_system->getMillis() > g_checksum_state->dialog->_lastEventPoll + 16) { 
+			g_checksum_state->dialog->_lastEventPoll = g_system->getMillis(); 
+			g_gui.processEvent(event, g_checksum_state->dialog);
+			g_system->updateScreen();
+		}
+	}
+
+	return true;
+};
+
+static bool progressUpdateCallback(GUI::IntegrityDialog *dialog, int bytesProcessed) {
```

Comment:
So, since it is now void, write like this:

```suggestion
static bool progressUpdateCallback(void *param, int bytesProcessed) {
     Dialog  *dialog = (GUI::IntegrityDialog *)param;
```